### PR TITLE
feat: bump actions to use latest node runtime

### PR DESCRIPTION
**Pull request by chenrui333**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -22,9 +22,9 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
-          node-version: "14"
+          node-version-file: .tool-versions
```
Comment (jiacai2050) [inline .github/workflows/lint.yml:27]: This is only required in CI, and should not affect devs

File: Makefile [hunk 2]
```diff
@@ -7,7 +7,7 @@ init:
 	git submodule update --init
 
 lint:
-	npx prettier@2.7.1 . --check
+	npx prettier . --check
```
Comment (jiacai2050) [inline Makefile:10]: I prefer stick to a fixed version, as prettier may change style.

File: .tool-versions
Comment (jiacai2050) [inline .tool-versions:1]: This is not required.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (chenrui333) [reply .github/workflows/lint.yml:27]: that is fine, I will bump it to the latest LTS version

File: .tool-versions
Comment (chenrui333) [reply .tool-versions:1]: 👍

File: Makefile (hunk 2 above)
Comment (chenrui333) [reply Makefile:10]: right now it is v3, so let's bump to the latest?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -22,9 +22,9 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
-          node-version: "14"
+          node-version: "20"
```
Comment (jiacai2050) [inline .github/workflows/lint.yml:27]: latest

Comment (jiacai2050) [inline .github/workflows/lint.yml:27]:
```suggestion
          node-version: "latest"
```

**Pull request merged**